File: python/src/sort/gnome_sort.py

def gnome_sort(in_array):
    curr, length = 0, len(in_array)
    nxt = curr + 1    
    while nxt < length:
        if in_array[curr] > in_array[nxt]:
            tmp = in_array[curr]
            in_array[curr] = in_array[nxt]
            in_array[nxt]  = tmp
            
            if curr > 0:
                curr -= 1
        else:
            curr += 1 

        nxt = curr + 1 
    return in_array

File: c#/Sort/GnomeSort.cs
using System.Collections;

namespace Sort
{
    public class GnomeSort : ISort
    {
        public ArrayList Sort(ArrayList inArray)
        {
            int current = 0;
            int next = current + 1;

            while (next < inArray.Count)
            {
                if( (int)inArray[current] > (int)inArray[next] )
                {
                    var tmp = inArray[current];
                    inArray[current] = inArray[next];
                    inArray[next] = tmp;

                    if (current > 0)
                        current--;
                }
                else
                {
                    current++;
                }

                next = current + 1;
            }

            return inArray;
        }
    }
}
